### DISCUSSION TITLE
RabbitMq version (#5.20.0)- DefaultConsumer nextDelivery() method is not available

### DISCUSSION THREAD

**Discussion by mageshmani18** [rabbitmq/rabbitmq-java-client#1212] [Q&A] [answered]
Hello,

The nextDelivery() method is not available, and it seems like deprecated in the latest version(#5.20.0) of rabbitmq. 

Please suggest me how to get the delivery using DefaultConsumer.

Thanks,
Magesh M

**Comment by michaelklishin** [comment, marked as answer]
As [demonstrated in the tutorials](https://rabbitmq.com/getstarted.html) and the [Java client API doc guide](https://rabbitmq.com/api-guide.html#consuming).

DO NOT use `QueueingConsumer` (a `Consumer` implementation that does provide `nextDelivery`), it is deprecated and will be removed in a later version. In fact, I am very surprised that we haven't removed it yet.

**Reply by acogoluegnes** [reply]
`QueueingConsumer` is in the [`test` directory](https://github.com/rabbitmq/rabbitmq-java-client/blob/5.x.x-stable/src/test/java/com/rabbitmq/client/QueueingConsumer.java), it should not be available in the JAR file. It's a handy class for testing, but it has limitations for production use.

It was removed from the main distribution in [5.0.0](https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.0.0) (September 2017).

**Reply by michaelklishin** [reply]
Ahhh, OK then :) Yes, it is OK for isolated tests but is too prone to cause JVM heap explosions under load with a high prefetch number (or in automatic acknowledgement modes).

I'm explaining this to those landing here from Google searches, not to Arnaud, of course :)
